code/common/datadog.q

```diff
@@ -0,0 +1,36 @@
+// DATADOG CHECKS
+
+//Create datadog namespace
+\d .dg
+
+ddconfigfile:@[value;`ddconfigfile;hsym `$getenv[`KDBAPPCONFIG],"/ddconfig.txt"]
+
+//sets dogstatsd_port to the port defined by ddconfigfile
+$[`ddconfig.txt in key hsym `$getenv[`KDBAPPCONFIG];value each read0 ddconfigfile;dogstatsd_port:8125]
+
+//Functions are set to return 1b or 0b based on the health of the service.
+//Default check returns 1b from each process to indicate process is up and can be queried.
+//These checks run every time crontab/timer is run.
+
+handlers:(`symbol$())!()
+
+is_ok:{[x]
+        f:$[@[{[x].proc.proctype in key .dg.handlers};`;0b];
+                .dg.handlers[.proc.proctype];
+                .dg.default_is_ok];
+        @[f;`;0b]
+        }
+
+default_is_ok:{[x]1b}
+
+//functions to send metrics and events to datadog from TorQ processes
+sendMetric:{[metric_name;metric_value] system"bash -c \"echo  -n '",metric_name,":",(string metric_value),"|g|#shell' > /dev/udp/127.0.0.1/",(string dogstatsd_port),"\"";}
+
+.dg.sendEvent:{[event_title;event_text;tags;alert_type] system"event_title=",event_title,"; event_text=","\"",event_text,"\"","; tags=","\"",$[0h=type tags;"#",("," sv tags);"#",tags],"\"",";alert_type=",alert_type,"; ","echo \"_e{${#event_title},${#event_text}}:$event_title|$event_text|#$tags|t:$alert_type\" |nc -4u -w0 127.0.0.1 ",string dogstatsd_port;}
```

Comment:
This is one of the biggest lines I've seen in a while, should be broken up for readability.
```suggestion
.dg.sendEvent:{[event_title;event_text;tags;alert_type]
  system"event_title=",event_title,"; event_text=","\"",event_text,"\"","; tags=","\"#",$[0h=type tags;","sv tags;tags],"\"",";alert_type=",alert_type,"; ","echo \"_e{${#event_title},${#event_text}}:$event_title|$event_text|#$tags|t:$alert_type\" |nc -4u -w0 127.0.0.1 ",string dogstatsd_port;
 }
```